index.js

```diff
@@ -0,0 +1,113 @@
+const express = require('express');
+const helmet = require('helmet');
+const knex = require("knex");
+const session = require('express-session');
+const KnexSessionStore = require('connect-session-knex')(session);
+const bcrypt = require("bcryptjs");
+const knexConfig = require("./knexfile");
+const cors = require("cors");
+
+const db = knex(knexConfig.development)
+
+const server = express();
+
+const sessionConfig = {
+    name: 'foobarbanana', 
+    secret: 'thisisasecretkey',
+    cookie: {
+      maxAge: 1000 * 60 * 10,
+      secure: false, // only set it over https; in production you want this true.
+    },
+    httpOnly: true, // no js can touch this cookie
+    resave: false,
+    saveUninitialized: false,
+    store: new KnexSessionStore({
+      tablename: 'sessions',
+      sidfieldname: 'sid',
+      knex: db,
+      createtable: true,
+      clearInterval: 1000 * 60 * 60,
+    }),
+  };
+
+server.use(session(sessionConfig));
+server.use(express.json());
+server.use(helmet());
+server.use(cors());
+
+const protected = (req, res, next) => req.session && req.session.user ? next() : res.status(401).json({ you: "shall not pass!"});
+
+server.post("/api/register", (req, res) => {
+    const creds = req.body;
+
+    if (creds.username === "" || creds.username === undefined) {
+        return res.status(400).json({error: "Please enter a username."})
+    }
+
+    if (creds.password === "" || creds.password === undefined) {
+        return res.status(400).json({error: "Please enter a password."})
+    }
+
+    const hash = bcrypt.hashSync(creds.password, 8);
+    creds.password = hash;
+    db("users")
+        .insert(creds)
+        .then(ids => res.status(201).json(ids))
+        .catch(err => res.status(401).json(err))
+})
+
+server.post("/api/login", (req, res) => {
```

Comment:
All of your endpoints look great.  With all of the repetition you are getting with writing them, they have become almost a habit.  Great work with your endpoints.  They are very readable